lib/tesla_api/result.ex

```diff
@@ -0,0 +1,38 @@
+defmodule TeslaApi.Result do
+  alias TeslaApi.Error
+  alias Tesla.Env
+
+  @spec handle({:ok | :error, Env.t() | atom}, (any -> any)) :: {:error, Error.t()} | {:ok, any}
+  def handle(result, transform \\ & &1)
+
+  def handle({:ok, %Env{status: status, body: %{"response" => response}}}, transform)
+      when status >= 200 and status <= 299 and is_list(response) do
+    {:ok, Enum.map(response, transform)}
+  end
+
+  def handle({:ok, %Env{status: status, body: %{"response" => response}}}, transform)
+      when status >= 200 and status <= 299 do
+    {:ok, transform.(response)}
+  end
+
+  def handle({:ok, %Env{status: 408, body: %{"error" => "vehicle unavailable:" <> _}} = e}, _) do
+    msg = "Vehicle unavailable. It must be woken up to make the API call succeed."
```

Comment:
This could probably be an `@` module variable.

Reply:
Yes, that's definitely an option. The other error messages should then also become module attributes, right? Just to be consistent. I believe it's more idiomatic to keep it inside the function though.

Reply:
I guess maybe it doesn't matter in the newer versions of the BEAM as they'll both reference the code rather than creating new data on the process's heap.